rb/lib/selenium/webdriver/remote/w3c/bridge.rb

```diff
@@ -358,13 +358,25 @@ def click_element(element)
             execute :element_click, id: element
           end
 
-          # TODO: - Implement file verification
+
           def send_keys_to_element(element, keys)
+            if @file_detector
+              local_file = @file_detector.call(keys)
+              keys = [upload(local_file)] if local_file
+            end
             # Keep .split(//) for backward compatibility for now
             text = keys.join('')
             execute :element_send_keys, {id: element}, {value: text.split(//), text: text}
           end
 
+          def upload(local_file)
+            unless File.file?(local_file)
+              raise Error::WebDriverError, "you may only upload files: #{local_file.inspect}"
```

Comment:
I'd alter this message to be a bit more verbose. Here you're taking an input which you know will only be called when file_detector is set. But your message says you may only upload files (Which is what you're doing)

Reply:
Sorry, but I not understand what you mean :) This check only show that local_file is file or not. How message should be verbose? 

I have only one idea. Add to message something like this: "File detector set, but return not file" or "File detector should return file". But in this case maybe this check should move after file_detector.call

P.S: This part of code I copied from oss bridge :)

Reply:
How about...

"You are trying to work with something that isn't a file." (Don't really need object inspect here)
"File detector only works with Files. #{local_file.inspect} isnt a file!" <- Obj inspect makes sense here.

P.s. if this is copied from OSS Bridge I'd probably fix it up there too.

Reply:
Got it. Fixed :)